torch/distributed/_tensor/op_schema.py

```diff
@@ -171,25 +190,80 @@ def __repr__(self) -> str:
             f" kwargs_schema={self.kwargs_schema})"
         )
 
+    def arg_type_tensor_or_tensor_list_like(self, arg_idx: int) -> bool:
+        op_arg_type = self.op._schema.arguments[arg_idx].type
+        is_tensor = isinstance(op_arg_type, torch.TensorType)
+        if is_tensor:
+            return True
+
+        is_list_like = isinstance(op_arg_type, torch.ListType)
+        if not is_list_like:
+            return False
+
+        elem_type = op_arg_type.getElementType()
+        return isinstance(elem_type, torch.TensorType) or (
+            isinstance(elem_type, torch.OptionalType)
+            and isinstance(elem_type.getElementType(), torch.TensorType)
+        )
```

Comment:
For my learning purpose, what cases will trigger this logic?

when `isinstance(elem_type, torch.TensorType)`

Reply:
for argument types like `Tensor[]` or `Tensor?[]`, this is basically to test these routes and make sure we don't hash those arguments